### PR TITLE
snippets: nus-console: Redirect Shell to nus-console

### PR DESCRIPTION
**Pull request by ubieda**
To allow easily building samples using the shell, without needing to specify additional configurations. Tested with:
```
west build -b nrf52840dk/nrf52840 \
           -S nus-console` \
           samples/subsys/shell/shell_module
```

### CODE REVIEW

**Comment by ubieda** [comment]
Unfortunately, the latest main is broken (`git bisect` points to https://github.com/zephyrproject-rtos/zephyr/pull/65071). However, I tried this patch from https://github.com/zephyrproject-rtos/zephyr/commit/92e0d0bbfc76ba41c3844b7362bcccf488895087 and works as expected.

**Comment by ithinuel** [comment]
Hello, would you mind testing on top of https://github.com/zephyrproject-rtos/zephyr/pull/71839?

**Review by jori-nordic — APPROVED** [review]
tested locally with [your script](https://github.com/ubieda/etales-zephyr/blob/master/scripts/bluetoothcli)

**Comment by ubieda** [comment]
> Hello, would you mind testing on top of #71839?

@ithinuel I tried after rebasing (main head: 2765d23775146c58b91c5602664308f777fb0346), which includes the PR you're referencing, and unfortunately the issue still persists.

Here's a snapshot I get from `west attach`, one second after `west flash --erase`:

``` gdb
GNU gdb (Zephyr SDK 0.16.4) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "--host=x86_64-build_pc-linux-gnu --target=arm-zephyr-eabi".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://github.com/zephyrproject-rtos/sdk-ng/issues>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/ubieda/zephyrproject/zephyr/build/zephyr/zephyr.elf...
Remote debugging using :2331
__enable_irq () at /home/ubieda/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include/cmsis_gcc.h:951
951	  __ASM volatile ("cpsie i" : : : "memory");
(gdb) bt
#0  __enable_irq () at /home/ubieda/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include/cmsis_gcc.h:951
#1  arch_cpu_idle () at /home/ubieda/zephyrproject/zephyr/arch/arm/core/cortex_m/cpu_idle.c:96
#2  0x00022c6c in k_cpu_idle () at /home/ubieda/zephyrproject/zephyr/include/zephyr/kernel.h:5851
#3  idle (unused1=<optimized out>, unused2=<optimized out>, unused3=<optimized out>) at /home/ubieda/zephyrproject/zephyr/kernel/idle.c:89
#4  0x00002962 in z_thread_entry (entry=0x22c59 <idle>, p1=0x20001a58 <_kernel>, p2=0x0 <zephyr_timer_wrapper>, p3=0x0 <zephyr_timer_wrapper>)
    at /home/ubieda/zephyrproject/zephyr/lib/os/thread_entry.c:48
#5  0xaaaaaaaa in ?? ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb) 

```

**Comment by ithinuel** [comment]
I don’t really know how to test this but as far as I a can tell, the application is running.

Without this change, `west build -b nrf52840dk/nrf52840 -S nus-console samples/subsys/shell/shell_module`:
- the uart shell is available on the CMSIS-DAP’s uart.
- I can connect to device using nRF Toolbox’s UART service but no output is shown

with this change, and the same build command:
- Nothing shows on the CMSIS-DAP’s uart
- connecting to the device’s UART service shows some warning/error messages but it’s a bit difficult to figure what they are saying, I didn’t dig into this.

**Review by nashif — APPROVED** [review]

**Comment by ubieda** [comment]
@ithinuel Thanks. I went back to bisect this issue and I agree the referenced PR does address the first issue I was having.

The outstanding issue (excessive err-logs) is a separate issue, which I bisected and captured the details on https://github.com/zephyrproject-rtos/zephyr/issues/72019.

I leave it up to the reviewers to decide whether we should block this one-liner PR until that is addressed.

**Review by jori-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
